test/unit/math/prim/prob/multi_student_t_cholesky_test.cpp

```diff
@@ -0,0 +1,461 @@
+  Matrix<double, Dynamic, 1> mu(3, 1);
+  mu << 2.0, 3.0, 11.0;
+
+  Matrix<double, Dynamic, Dynamic> s(3, 3);
+  s << 10.0, 3.0, 11.0, 3.0, 9.0, 1.2, 11.0, 1.2, 16.0;
+  Matrix<double, Dynamic, Dynamic> L = s.llt().matrixL();
+  int N = 10000;
+  int K = stan::math::round(2 * std::pow(N, 0.4));
+  boost::math::students_t_distribution<> dist(3.0);
+  boost::math::chi_squared mydist(K - 1);
+
+  double loc[K - 1];
+  for (int i = 1; i < K; i++)
+    loc[i - 1] = quantile(dist, i * std::pow(K, -1.0));
+
+  int count = 0;
+  int bin[K];
+  double expect[K];
+  for (int i = 0; i < K; i++) {
+    bin[i] = 0;
+    expect[i] = N / K;
+  }
+
+  Eigen::VectorXd a(mu.rows());
+  while (count < N) {
+    a = multi_student_t_cholesky_rng(3.0, mu, L, rng);
+    a(1) = (a(1) - mu(1, 0)) / std::sqrt(s(1, 1));
+    int i = 0;
+    while (i < K - 1 && a(1) > loc[i])
+      ++i;
+    ++bin[i];
+    count++;
+  }
+
+  double chi = 0;
+  for (int j = 0; j < K; j++)
+    chi += ((bin[j] - expect[j]) * (bin[j] - expect[j]) / expect[j]);
+
+  EXPECT_TRUE(chi < quantile(complement(mydist, 1e-6)));
+}
+
+TEST(ProbDistributionsMultiStudentTCholesky, WrongSize) {
+  using Eigen::Dynamic;
+  using Eigen::Matrix;
+  using stan::math::multi_student_t_cholesky_lpdf;
+  using std::vector;
+  vector<Matrix<double, Dynamic, 1> > y_3_3(3);
+  vector<Matrix<double, Dynamic, 1> > y_3_1(3);
+  vector<Matrix<double, Dynamic, 1> > y_3_2(3);
+  vector<Matrix<double, Dynamic, 1> > y_1_3(1);
+  vector<Matrix<double, Dynamic, 1> > y_2_3(2);
+  Matrix<double, Dynamic, 1> y_3(3);
+  Matrix<double, Dynamic, 1> y_2(2);
+  Matrix<double, Dynamic, 1> y_1(1);
+  y_3 << 2.0, -2.0, 11.0;
+  y_2 << 2.0, -2.0;
+  y_1 << 2.0;
+  y_3_3[0] = y_3;
+  y_3_3[1] = y_3;
+  y_3_3[2] = y_3;
+  y_3_1[0] = y_1;
+  y_3_1[1] = y_1;
+  y_3_1[2] = y_1;
+  y_3_2[0] = y_2;
+  y_3_2[1] = y_2;
+  y_3_2[2] = y_2;
+  y_1_3[0] = y_3;
+  y_2_3[0] = y_3;
+  y_2_3[1] = y_3;
+
+  vector<Matrix<double, Dynamic, 1> > mu_3_3(3);
+  Matrix<double, Dynamic, 1> mu_3(3);
+  mu_3 << 2.0, -2.0, 11.0;
+  mu_3_3[0] = mu_3;
+  mu_3_3[1] = mu_3;
+  mu_3_3[2] = mu_3;
+
+  Matrix<double, Dynamic, Dynamic> Sigma(3, 3);
+  Sigma << 10.0, -3.0, 0.0, -3.0, 5.0, 0.0, 0.0, 0.0, 5.0;
+  Matrix<double, Dynamic, Dynamic> L = Sigma.llt().matrixL();
+
+  double nu = 4.0;
+
+  EXPECT_NO_THROW(
+      stan::math::multi_student_t_cholesky_lpdf(y_3_3, nu, mu_3_3, L));
+  EXPECT_NO_THROW(
+      stan::math::multi_student_t_cholesky_lpdf(y_3, nu, mu_3_3, L));
+
+  EXPECT_THROW(stan::math::multi_student_t_cholesky_lpdf(y_1_3, nu, mu_3_3, L),
+               std::invalid_argument);
+  EXPECT_THROW(stan::math::multi_student_t_cholesky_lpdf(y_2_3, nu, mu_3_3, L),
+               std::invalid_argument);
+  EXPECT_THROW(stan::math::multi_student_t_cholesky_lpdf(y_3_1, nu, mu_3_3, L),
+               std::invalid_argument);
+  EXPECT_THROW(stan::math::multi_student_t_cholesky_lpdf(y_3_2, nu, mu_3_3, L),
+               std::invalid_argument);
+}
```

Comment:
We also need tests of boundary conditions like size 0 inputs.  Did I miss those?  You basically want to try to test all the branch points that return in the code.

Reply:
These are all checked. Ctrl + F all these instances `std::numeric_limits<double>::quiet_NaN();`